src/function_wrappers.ts

```diff
@@ -122,6 +131,37 @@ const wrapHttpFunction = (execute: HttpFunction): RequestHandler => {
   };
 };
 
+const wrapHttpAsyncFunction = (
+  userFunction: OpenFunction,
+  context: OpenFunctionContext
+): RequestHandler => {
+  const ctx = OpenFunctionRuntime.ProxyContext(context);
+  const httpHandler = (req: Request, res: Response) => {
+    const callback = getOnDoneCallback(res);
+
+    Promise.resolve()
+      .then(() => userFunction(ctx, req.body))
+      .then(result => {
+        debug('ℹ️ User function returned: %j', result);
+
+        const data = get(result, 'body');
+        const code = get(result, 'code', 200);
+        const headers = get(result, 'headers');
+
+        !isEmpty(headers) && res.set(headers);
+
+        if (data !== undefined) {
+          res.status(code).send(data);
```

Comment:
`res.status(code).send(data); ` the send here is to send response data back to caller via HTTP?
The same data is send to dapr outputs?

If so maybe we needn't send the data back to caller via HTTP again? because the dapr outputs already got these data

Reply:
Here we leave this question, whether to send anything back via HTTP, open to the function developer. If no data was returned from `func(ctx, data)`, thus nothing would be send back.

Reply:
Got your point in previous comment, we can do this given that `req` and `res` are passed to context.